.github/workflows/post-merge-validation.yml

```
@@ -0,0 +1,145 @@
+  workflow_dispatch:  # Allow manual triggering
+
+jobs:
+  validate-merge:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0  # Get full history for comparisons
+      
+      - name: Set up Python
+        uses: actions/setup-python@v4
```

actionlint (1.7.7)

the runner of "actions/setup-python@v4" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)